packaging/rpm/microshift.spec

```diff
@@ -729,10 +768,25 @@ fi
 %files ai-model-serving-release-info
 %{_datadir}/microshift/release/release-ai-model-serving-x86_64.json
 
+%files observability
+%dir %{_prefix}/lib/microshift/manifests.d/003-microshift-observability
+%dir %{_sharedstatedir}/microshift-observability
+%{_unitdir}/microshift-observability.service
+%{_presetdir}/90-enable-microshift-observability.preset
```

Comment:
Why do we need this preset?

Reply:
If the observability rpm were installed without the preset, then it's disabled by default and `systemctl enable microshift` command doesn't propagate to dependencies. That, coupled with the service settings `RequiredBy=microshift.service`, means that microshift itself can't start without the user manually enabling the observability service.

This way, the  user isn't required to manage the service, as it's handled automatically.

Reply:
What about the other way around? When microshift+observability are installed, microshift is disabled, but observability enabled by default. So, the observability service will fail on reboot.

Reply:
I think we need to implement the start / stop dependency in a different manner. 
Did we consider using the following configuration for the observability service?
```
[Unit]
PartOf=microshift.service
After=microshift.service
```

Reply:
>Isn't it enough to couple service start / stop?

Well, no. In order for the `RequiredBy=` directive to take effect, the service must be `enabled`. But as you said, if it's enabled, then after a reboot it would start and enter a fail loop. This is fixed by applying the `After=microshift.service`.

So the .preset file + the `RequiredBy=` is what activates the dependency. The `After=` ensures that the service doesn't start until microshift is active. `PartOf=` propagates stops/restarts of the microshift service to the observability service.

But I think in the end this ended up being a long walk to accomplishing (basically) the same thing as adding `microshift-observability` to the `microshift.service`'s `Wants=` directive.